jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1460,18 +1460,23 @@ void java_object_factoryt::gen_nondet_array_init(
 ///     expr = $VALUES[i];
 /// where $VALUES is a variable generated by the Java compiler that stores
 /// the array that is returned by Enum.values().
-void java_object_factoryt::gen_nondet_enum_init(
+/// This may fail if the $VALUES static field is not present, which gives the
+/// possible values of a particular enum subtype. Ensuring that field is in the
```

Comment:
:pick: "which gives the..." <- a bit confusing to have the explanation of what `$VALUES` is split like that, maybe add this part to the previous sentence?

Reply:
Fixed this and added a warning message